src/Adapter/AttributeGroup/QueryHandler/GetAttributeGroupNameByIdHandler.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\AttributeGroup\QueryHandler;
+
+use PrestaShop\PrestaShop\Adapter\AttributeGroup\AbstractAttributeGroupHandler;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Query\GetAttributeGroupNameById;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\QueryHandler\GetAttributeGroupNameByIdHandlerInterface;
+
+/**
+ * Handles query which gets attribute group name by id using legacy object model
+ */
+final class GetAttributeGroupNameByIdHandler extends AbstractAttributeGroupHandler implements GetAttributeGroupNameByIdHandlerInterface
```

Comment:
Maybe `GetAttributeGroupName` would be enough? @matks we are starting to introduce a lot of generic queries that return single value, is that ok?

Reply:
Yes I agree with @sarjon here

This query can contain today a single `private $attributeGroupId;` and in the future we could add new properties with new ways of filtering